update-motd.d/88-esm-announce

```diff
@@ -1,4 +1,37 @@
 #!/bin/sh
-stamp="/var/lib/ubuntu-advantage/messages/motd-esm-announce"
 
-[ ! -r "$stamp" ] || cat "$stamp"
+if [ -f /etc/apt/sources.list.d/ubuntu-esm-apps.list ]; then
+    # esm-apps already enabed, no need to announce it
+    exit
+fi
+if lsb_release -ds | grep -qv "LTS"; then
+    # not on LTS release, no need to announce esm-apps
+    exit
+fi
+
+# determine contextual url
+url="https://ubuntu.com/pro"
+codename=$(lsb_release -cs)
+if [ "$codename" = "xenial" ]; then
+    if [ -f "/run/cloud-init/cloud-id-azure" ]; then
+        url="https://ubuntu.com/16-04/azure"
+    else
+        url="https://ubuntu.com/16-04"
+    fi
+else
+    if [ -f "/run/cloud-init/cloud-id-aws" ]; then
+        url="https://ubuntu.com/aws/pro"
+    elif [ -f "/run/cloud-init/cloud-id-azure" ]; then
+        url="https://ubuntu.com/azure/pro"
+    elif [ -f "/run/cloud-init/cloud-id-gce" ]; then
+        url="https://ubuntu.com/gcp/pro"
+    fi
+fi
+
+echo
+echo " * Introducing Expanded Security Maintenance for Applications."
+echo "   Receive updates to over 25,000 software packages with your"
+echo "   Ubuntu Pro subscription. Free for personal use."
+echo
+echo "     $url"
```

Comment:
@lechsandecki How long do we want to keep this message around?
```
* Introducing Expanded Security Maintenance for Applications.
  Receive updates to over 25,000 software packages with your
  Ubuntu Pro subscription. Free for personal use.
```

Reply:
good call, i think it's time to drop it since we also have apt-news and apt-hooks

Reply:
Sounds good - thanks!